app/services/account_search_service.rb

```diff
@@ -133,6 +132,36 @@ def time_distance_function
     }
   end
 
+  def must_clause
+    fields = %w(username username.* display_name display_name.*)
+    fields << 'text' << 'text.*' if options[:use_searchable_text]
+
+    [
+      {
+        multi_match: {
+          query: terms_for_query,
+          fields: fields,
+          type: 'best_fields',
+          operator: 'or',
+        },
+      },
+    ]
+  end
+
+  def should_clause
+    [
+      {
+        multi_match: {
+          query: terms_for_query,
+          fields: %w(username username.* display_name display_name.*),
+          type: 'best_fields',
+          operator: 'and',
+          boost: 10,
+        },
+      },
+    ]
+  end
```

Comment:
Since this drops any reference to `acct`, I'm worried this would disadvantage auto-completion of remote user mentions when you start typing their full handle, e.g. `@gargron@ma` should match `@gargron@mastodon.social` rather than `@gargron@fake.server`

Reply:
It’s just a field rename. Username is still both username and domain.

Reply:
Ah, right, I missed that! Not sure why the definition doesn't simply reuse `acct` or `pretty_act` though.